.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,110 @@
+name: Build and Test Rust Programs
+
+on:
+  push:
+    branches: [master]
+    paths:
+      - 'rust/**'
+  pull_request:
+    branches: [master]
+    paths:
+      - 'rust/**'
+
+env:
+  CARGO_TERM_COLOR: always
+  SOLANA_VERSION: 1.8.2
```

Comment:
It would be nice to automatically test using the currently active versions of Solana on mainnet. I'm looking around for this info now

Reply:
In that case we should create a reusable action to install `stable|beta` versions of solana.
We can look into that and update the workflow later, shouldn't block the PR though IMO to get CI fixed at least.